src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -153,26 +155,106 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
             CancellationToken cancellationToken)
         {
 
+            
             DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
                 manifestId,
                 targetDirectory,
                 proxyUri: null,
                 minimatchPatterns: null);
-            using( Process tarring = new Process())
+            
+            var processTcs = new TaskCompletionSource<int>();
+
+            using(var cancelSource = new CancellationTokenSource())
+            using(var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cancelSource.Token))
+            using(var process = new Process())
             {
-                tarring.StartInfo.FileName = "tar";
-                tarring.StartInfo.Arguments = $"-xf - -C {targetDirectory}";
-                tarring.StartInfo.UseShellExecute = false;
-                tarring.StartInfo.RedirectStandardInput = true;
-                tarring.Start();
-                return dedupManifestClient.DownloadPipelineCacheAsync(options, cancellationToken, tarring.StandardInput.BaseStream);
+                process.StartInfo.FileName = @"C:\Program Files\7-Zip\7z.exe"; // tar // @"C:\Program Files\7-Zip\7z.exe"
+                process.StartInfo.Arguments = $"x -si -aoa -o{targetDirectory} -ttar"; // -xf - -C {targetDirectory} // tarring.StartInfo.Arguments = $"x -si -aoa -o{targetDirectory} -ttar";
+                process.StartInfo.UseShellExecute = false;
+                process.StartInfo.RedirectStandardInput = true;
+                process.StartInfo.RedirectStandardOutput = true;
+                process.StartInfo.RedirectStandardError = true;
+                process.EnableRaisingEvents = true;
+                process.Exited += (sender, args) =>
+                {
+                    cancelSource.Cancel();
+                    processTcs.SetResult(process.ExitCode);
+                };
+
+                context.Info($"Starting '{process.StartInfo.FileName}' with arguments '{process.StartInfo.Arguments}'...");
```

Comment:
This feels more like a `Debug` message versus an `Info` message ... most users don't need to know/care how the content of the cache is getting zipped/unzipped.

Reply:
Agree.